### ISSUE TITLE
Add `Checkout and Slide In` action to git4idea action group(s)

### ISSUE THREAD

**Issue by mkondratek** [VirtusLab/git-machete-intellij-plugin#1173]
![image](https://user-images.githubusercontent.com/19799111/192353705-81b51196-685b-435a-85bb-48a3284e8791.png)

I would like to checkout `origin/move-to-previous-next-parent` and have it in my branch layout. 
I think the most convenient solution is to add an action to git4idea log context menu.
Also, DVCS widget could be considered and the action might be added there too.

**Comment by PawelLipski** [comment]
Out of curiosity, to what extent is your proposed feature related to #1152? or can the two be introduced independently from each other?

**Comment by mkondratek** [comment]
#1152 will provide a workaround but I would not call it a solution. 

We want to checkout and slide in a branch.
With #1152, it can be done inderectly - first we are checking out a branch, then slide in with the notification.
For me it is like we suggest a fix to some ambigous situation. There is multiple ways to change a brach and get the notification.

With this one, it is direct - checkout and slide in a specific non managed branch. 
Also, the place is precise (git4idea context menu/dvcs widget branch context menu).

IMHO they are different.
Another approach would be to autocomplete remote branches and correctly handle them with slide in dialog. 
I have not been thinking about it for long but at the first glance it is a bit more complicated.
Probably I am used to my specific way of checking out remote branches.

**Comment by mkondratek** [comment]
I ve checked the slide in of exsiting remote branch - works nice and I think we can add it...
BUT it requires to specify a branch under which we want to slide in a branch and I think thats what makes this approach is more complicated.

**Comment by PawelLipski** [comment]
> under which we want to slide in a branch

Note that this can be inferred automatically, using the same algorithm that underlies Discover... ofc this algorithm is somewhat imperfect, the guess might always be different from what the user expected

**Comment by mkondratek** [comment]
I agree but that's a new action - `Slide In` works with a specific branch (often indicated with selection and right click). Some toolbar action `Auto Slide In (?)` could work as you described.
